### PR TITLE
COMPLETELY fix canvas freeze

### PR DESCRIPTION
**Pull request by Hamms**
See https://github.com/code-dot-org/dance-party/issues/514#issuecomment-443369660

### CODE REVIEW

File: src/replayToMovie.js [hunk 1]
```diff
@@ -81,6 +79,46 @@ const p5Inst = new P5(function (p5obj) {
...
+  p5Inst._renderer.scale = function(x, y) {
+    if (x === 0) {
+      x = 0.001;
+    }
+
+    if (y === 0) {
+      y = 0.001;
+    }
+
+    return origScale.call(this, x, y);
```
Comment (islemaster) [inline src/replayToMovie.js]: Note: Hopefully irrelevant to our use case, but [`p5.scale()`](https://p5js.org/reference/#/p5/scale) supports a few other call patterns we should make sure we don't depend on anywhere.

- `scale([x, y])` is probably not common, but may be worth spot-checking Effects.js.
- `scale(createVector(x, y))` probably a non-issue for us.
- `scale(x, y, z)` should only be relevant when using WebGL so not an issue for us.

File: src/replayToMovie.js [hunk 2]
```diff
@@ -81,6 +79,46 @@ const p5Inst = new P5(function (p5obj) {
   p5obj.height = HEIGHT;
 });
 
+createNewP5RenderingCanvas();
```
Comment (islemaster) [inline src/replayToMovie.js]: Why do we call this statically here (and not keep the reference to the created canvas)?

File: src/replayToMovie.js (hunk 1 above)
Comment (Hamms) [reply src/replayToMovie.js]: Alternatively, I could apply this override to [canvas.scale()](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/scale) instead. I had been considering doing that anyway

File: src/replayToMovie.js (hunk 2 above)
Comment (Hamms) [reply src/replayToMovie.js]: For silly reasons; the Effects constructor uses `p5.rgb` in a couple places, which internally references `p5._renderer`, so we need to have one created before initializing the effects.

I'll add a comment to this effect

**Comment by Hamms** [comment]
PTAL

**Pull request merged**